Update to diesel 0.99 and r2d2 0.8.1

ChristophWurst opened the pull request:
This update Diesel to v0.99 and r2d2 to v0.8.1 (both latest available). I had to remove the configuration option on the middleware constructor as that [has been removed from the r2d2 lib](https://github.com/sfackler/r2d2/blob/06a52109609a1cbf5771053250e5574cc5e644dc/CHANGELOG.md#changed).

bradleybeddoes commented:
Bugger, the ability to customise the `Pool` is something I'd really like to maintain for folks.

Thoughts on a `DieselMiddlewareBuilder` with a method that takes a `r2d2::Pool` instance for use within the constructed `DieselMiddleware`?

cc @smangelsdorf

bradleybeddoes commented:
Also how about we jump straight up to [`1.0.0-beta1`](https://github.com/diesel-rs/diesel/blob/master/CHANGELOG.md) ?

ChristophWurst commented:
@bradleybeddoes how about one simple `new()` constructor method and one that allows to pass in a custom pool like `with_pool(…)`?

> Also how about we jump straight up to 1.0.0-beta1 ?

Makes sense.

bradleybeddoes commented:
@smangelsdorf and I chatted on this earlier, we're good to go with your suggestion of:

``` rust
new(database_url: &str) -> Self
with_pool(database_url: &str, pool: Pool::<ConnectionManager<T>>) -> Self
```

ChristophWurst commented:
> @smangelsdorf and I chatted on this earlier, we're good to go with your suggestion of:

Done. Note that when passing in a pool instance, the `database_url` has already been set when constructing the `ConnectionManager`, thus this parameter is useless and I chose to remove it from the interface.

I've added two tests that cover the construction of a middleware with default and custom configuration. Since the db connection is tried to be established right away, those tests fail without a running db. So I added the `#[ignore]` attribute and those tests can be executed with `cargo test -- --ignored`.

smangelsdorf commented on src/lib.rs:
```diff
@@ -63,20 +63,32 @@ where
+
+        DieselMiddleware {
+            pool: AssertUnwindSafe(pool),
+        }
```
Could this now be implemented using `with_pool`? 😁

ChristophWurst replied:
makes sense 👍

bradleybeddoes commented:
I like the idea of the tests but I’m thinking an in memory SQLite instance might make life easier as opposed to needing a Postgres instance during test runs.

Thoughts?

ChristophWurst commented:
While I'm not a big fan of SQLite, it's fine to use it for this purpose. I've pushed another commit that addresses this change.

Let me know if you prefer to have one single commit for this kinf of change, than I can squash my commits.

bradleybeddoes commented:
Thanks for this PR @ChristophWurst, really good result here
